Scripts/Audio.cs

```diff
@@ -102,6 +102,10 @@ public class SpeechOptions
     {
         [SerializeField] private AudioOutput _audioOutput;
         [SerializeField] private Encoding _encoding;
+        /// <summary>
+        /// Id of the receiving microphone can be different on different devices.
+        /// </summary>
+        [SerializeField] private int _microphoneId;
```

Comment:
It would be better if we don't change `SpeechOptions` as this is part of the Charisma schema - instead could we add `microphoneId` as a parameter to `StartSpeechRecognition`, so something like:

```c#
playthrough.StartSpeechRecognition(gameObject, "my-microphone");
```

In this case we should also guard against `StartSpeechRecognition` being called multiple times in a row without `StopSpeechRecognition` being called. In this case, my thoughts are it should say something like...

```
`SpeechRecognition` is already started. To change microphone input, stop the speech recognition and start it with a different microphone ID.
```